src/views/topics/GdpTopic06View.vue

```diff
@@ -1,17 +1,237 @@
 <template>
-    <TopicMeta :topic="topic" />
+  <TopicMeta :topic="topic" />
+  <div id="Pascal-Brainchon-wrapper">
+    <canvas id="Pascal-Brainchon-canvas" width="500" height="500" />
+  </div>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
 import { indexTopicMap } from '@/data';
 import { Topic } from '@/types';
+import { fabric } from 'fabric';
+import { makeLabel, makeLine, makeMovablePoint } from '@/utils/canvas';
+import { IEvent } from 'fabric/fabric-impl';
+import { calculateLineIntersectInLinearEquation, solveLinearEquation } from '@/utils/geometry';
+
 const topic = indexTopicMap.get(6) as Topic;
+
+/**
```

Comment:
Are these needed? Why not just use types from fabric?

Reply:
Yes.
1. `Circle` and `Line` are subtypes of their fabric counterparts. For simplicity and readability I need to add more properties to those two types and subtyping is so far the best way I found to solve this.
2. `Intersection` should also be a subtype of `fabric.Intersection`. Good catch~
3. `fabric.Intersection` also has an outdated problem. As indicated by source code t should have a `points` property, but its type declaration in TS doesn't. So I have to subtype it to add this property. Maybe I should open an issue for them.